packages/2d/src/components/Polygon.ts

```diff
@@ -0,0 +1,104 @@
+import {Shape, ShapeProps} from './Shape';
+import {initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+export interface PolygonProps extends ShapeProps {
+  /**
+   * {@inheritDoc Polygon.nSides}
+   */
+  nSides?: SignalValue<number>;
+
+  /**
+   * {@inheritDoc Polygon.angle}
+   */
+  angle?: SignalValue<number>;
```

Comment:
I might be missing something here, but doesn't this end up doing basically the same as `rotation` already does? I understand that it draws the polygon at a different rotation without actually changing the `rotation` property of the element. But I'm not sure if that doesn't end up making things more confusing as there are now two ways of (visually) rotating a polygon. `Rect` also doesn't have an `angle` property.

Unless there's some use case I am overlooking?

Reply:
The use case I have in mind where this might matter is where you have children of the polygon.  Imagine you want to put text inside the polygon.  If you *rotate* the polygon to get a certain alignment, then the children will also rotate.  So then do you have to anti-rotate the children to get the text to align properly?
Similarly, the use case where you want to draw a connecting line from the polygon to another element, let's say from the right edge.  If you change the rotation of the polygon, then I think the right edge will also rotate, so the connecting point may no longer be where you expect.
I may well be wrong in my understanding of how child components work though, so open to illumination.

Reply:
Here is an example of the use case I have in mind.

```
  view.add(
    <>
      <Polygon 
        ref={hex} x={0} y={0} width={320} height={320} nSides={3} 
        stroke={'#fff'} lineWidth={8}
        angle={1 * 360/3}
        // rotation={1 * 360/3}
        fill={'orange'}
        children={<Txt ref={label} text="Yield" fill="white" />}
      />
    </>
  );

  yield* waitFor(0.5);

  yield* all(
    hex().nSides(8, 5),
    hex().fill("red", 5),
    hex().angle(0.5 * 360 / 8, 5),
    // hex().rotation(0.5 * 360 / 8, 5),
    label().text("Stop", 5)
  );
  
  yield* waitFor(0.5);
```

Using `angle`:

![motion-canvas-polygon-angle](https://user-images.githubusercontent.com/479998/222712988-8d15f6c4-9be4-4448-b98c-9400fe2fe8e6.gif)

Using `rotation`:

![motion-canvas-polygon-rotation](https://user-images.githubusercontent.com/479998/222713032-7d02d06c-00f9-4599-a93e-07ff8b84e676.gif)

Reply:
Would it help if I add this to the documentation?

```
   * @remarks
   * Use `angle` to change the apparent rotation of the polygon.
   * One reason you would use `angle` instead of `rotation` is when you
   * use child components, e.g. `Txt` inside the polygon.  In this case you
   * may want to change the angle of the polygon, rather than rotation of the 
   * polygon as well as it's children.
```

Reply:
I think I agree with @ksassnowski here
You can achieve the same thing by doing:
```tsx
<Node>
  <Polygon rotation={30} />
  <Txt>Some text here</Txt>
</Node>
```
This touches on a more general problem: transforming the content rendered by a node separately from its children. And I believe it should be addressed in the base `Node` class in a separate issue. This way other nodes like `Rect`s and `Img`s would also benefit from this.

When it comes to drawing lines between the polygon and other things, it could be more useful to add a `getVertexPosition(n)` method that would return the position of the vertex in local space.